src/main/java/org/broadinstitute/hellbender/tools/walkers/filters/VariantFiltration.java

```diff
@@ -0,0 +1,389 @@
+     * (e.g. if masking against Hapmap, use -maskName=hapmap for the normal masking and -maskName=not_hapmap for the reverse masking).
+     */
+    @Argument(fullName="filterNotInMask", shortName="filterNotInMask", doc="Filter records NOT in given input mask.", optional=true)
+    public boolean filterRecordsNotInMask = false;
+
+    /**
+     * By default, if JEXL cannot evaluate your expression for a particular record because one of the annotations is not present, the whole expression evaluates as PASSing.
+     * Use this argument to have it evaluate as failing filters instead for these cases.
+     */
+    @Argument(fullName="missingValuesInExpressionsShouldEvaluateAsFailing", doc="When evaluating the JEXL expressions, missing values should be considered failing the expression", optional=true)
+    public Boolean failMissingValues = false;
+
+    /**
+     * Invalidate previous filters applied to the VariantContext, applying only the filters here
+     */
+    @Argument(fullName="invalidatePreviousFilters",doc="Remove previous filters applied to the VCF",optional=true)
+    boolean invalidatePreviousFilters = false;
+
+    /**
+     * Invert the selection criteria for --filterExpression
+     */
+    @Argument(fullName="invertFilterExpression", shortName="invfilter", doc="Invert the selection criteria for --filterExpression", optional=true)
+    public boolean invertFilterExpression = false;
+
+    /**
+     * Invert the selection criteria for --genotypeFilterExpression
+     */
+    @Argument(fullName="invertGenotypeFilterExpression", shortName="invG_filter", doc="Invert the selection criteria for --genotypeFilterExpression", optional=true)
+    public boolean invertGenotypeFilterExpression = false;
+
+    /**
+     * If this argument is provided, set filtered genotypes to no-call (./.).
+     */
+    @Argument(fullName="setFilteredGtToNocall", optional=true, doc="Set filtered genotypes to no-call")
+    private boolean setFilteredGenotypesToNocall = false;
+
+    // JEXL expressions for the filters
+    private List<JexlVCMatchExp> filterExps;
+    private List<JexlVCMatchExp> genotypeFilterExps;
+
+    public static final String CLUSTERED_SNP_FILTER_NAME = "SnpCluster";
+
+    private VariantContextWriter writer;
+
+    private final List<Allele> diploidNoCallAlleles = Arrays.asList(Allele.NO_CALL, Allele.NO_CALL);
+
+    /**
+     * Prepend inverse phrase to description if --invertFilterExpression
+     *
+     * @param description the description
+     * @return the description with inverse prepended if --invert_filter_expression
+     */
+    private String possiblyInvertFilterExpression( final String description ){
+        return invertFilterExpression ? "Inverse of: " + description : description;
+    }
+
+    private void initializeVcfWriter() {
+
+        //XXX hardwired to output VCFs
+        //XXX unset indexing on the fly
```

Comment:
These `XXX` comments everywhere are ugly and might cause hellbender sources to appear in...."unrelated" search results. If there is a TODO, write TODO explicitly, otherwise write WARNING or similar